### PR TITLE
Remove LegacyAnnotation and [most] MoultingYaml

### PR DESCRIPTION
**Pull request by jackkoenig**
It has been deprecated since 1.1

This felt good. I couldn't fully delete the moultingyaml dependency because it's still used in memlib. I think json4s actually supports YAML so it may be possible to switch the code without breaking or deprecating the yaml support there.

### Contributor Checklist

- [x] Did you add Scaladoc to every public function/method?
- [x] Did you update the FIRRTL spec to include every new feature/behavior?
- [x] Did you add at least one test demonstrating the PR?
- [x] Did you delete any extraneous printlns/debugging code?
- [x] Did you specify the type of improvement?
- [x] Did you state the API impact?
- [x] Did you specify the code generation impact?
- [x] Did you request a desired merge strategy?
- [x] Did you add text to be included in the Release Notes for this change?

#### Type of Improvement

  - code cleanup   

#### API Impact

This removes support for `LegacyAnnotation` and related `YAML` stuff. Some of it is baked in to the `Driver` and related stuff which I left since those are set to be deleted elsewhere, I just elevated some warnings to "errors" although they aren't fatal.

#### Backend Code Generation Impact

No change

#### Desired Merge Strategy

  - Squash

#### Release Notes

Remove deprecated `LegacyAnnotation` and related YAML support. Also removed deprecated implicit annotation file handling.

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you add the appropriate labels?
- [ ] Did you mark the proper milestone (1.2.x, 1.3.0, 1.4.0) ?
- [ ] Did you review?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you mark as `Please Merge`?

### CODE REVIEW

**Comment by jackkoenig** [comment]
Requires https://github.com/freechipsproject/chisel3/pull/1544 to pass the chisel3 master tests

File: src/main/scala/firrtl/annotations/Annotation.scala [hunk 1]
```diff
@@ -127,130 +127,7 @@ trait MultiTargetAnnotation extends Annotation {
...
-            "This functionality will soon be removed. Please migrate to new annotations."
-          StageUtils.dramaticWarning(msg)
-          warned = true
-        }
-        annox
-      case other => other
-    }
-  }
-}
+object Annotation
```
Comment (seldridge) [inline src/main/scala/firrtl/annotations/Annotation.scala:130]: How would we go about removing this object?

Comment (jackkoenig) [reply src/main/scala/firrtl/annotations/Annotation.scala:130]: I don't know why anyone would be relying on it, but we'd have to deprecate. I'm not sure that we should though, it's a reasonable place to put utilities or implicit conversions if needed.

Comment (seldridge) [reply src/main/scala/firrtl/annotations/Annotation.scala:130]: Sounds good. Mostly just a curiosity... Yes, this could be reuse for utilities later.

**Pull request merged**